packages/hooks/src/createRunHook.js

```diff
@@ -60,6 +60,8 @@ function createRunHook( hooks, storeKey, returnFirstArg = false ) {
 		if ( returnFirstArg ) {
 			return args[ 0 ];
 		}
+
+		return undefined;
```

Comment:
TS was complaining about this code path not returning anything, so I just added `return undefined` to satisfy it. Interestingly, TS is supposed to be _better_ at handling this scenario in TS 5.1, but I suppose that improvement might not apply if the types are in JSDoc.